src/typeahead/typeahead.ts

```diff
@@ -284,13 +285,20 @@ export class NgbTypeahead implements ControlValueAccessor,
       this._windowRef.instance.selectEvent.subscribe((result: any) => this._selectResultClosePopup(result));
       this._windowRef.instance.activeChangeEvent.subscribe((activeId: string) => this.activeDescendant = activeId);
 
+      const nativeElement = this._windowRef.location.nativeElement;
+      this._mousedownRemoveListener =
+          this._renderer.listen(nativeElement, 'mousedown', (event) => { event.preventDefault(); });
+
       if (this.container === 'body') {
-        window.document.querySelector(this.container).appendChild(this._windowRef.location.nativeElement);
+        window.document.querySelector(this.container).appendChild(nativeElement);
```

Comment:
you should `@Inject(DOCUMENT) private _document: any`, not use `window.document`

Reply:
Oh, see it was like that before, but is worth fixing anyway!